### DISCUSSION TITLE
Alternative Inference Engines (LlamaFile, PowerInfer)

### DISCUSSION THREAD

**Discussion by BradKML** [khoj-ai/khoj#865] [Ideas]
Not sure if those are just hype or not, but LLamaFile recently noted that they found a way to speed up CPU inference. Would be nice to do fast Chatbots with docs on laptop.

**Comment by debanjum** [comment]
Given [LlamaFile exposes an OpenAI API compatible proxy](https://github.com/Mozilla-Ocho/llamafile?tab=readme-ov-file#json-api-quickstart). You can have Khoj use any Llamafile based chat model. See the [Khoj OpenAI Proxy Docs](https://docs.khoj.dev/advanced/use-openai-proxy) for how to set it up

_P.S: Not sure how I missed replying before, my notification for discussion topics maybe wonky_

**Reply by BradKML** [reply]
Thanks for the help and the self-host support
